src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -113,6 +113,38 @@ maxLength switch
                 .WithDescription("Combines multiple arrays and returns the concatenated array.")
                 .WithVariableParameter("arg", LanguageConstants.Array, minimumCount: 1, "The array for concatenation")
                 .Build();
+            
+            yield return new FunctionOverloadBuilder("toLogicalZone")
+                .WithReturnType(LanguageConstants.String)
+                .WithGenericDescription("Returns the logical zone corresponding to the given physical zone.")
+                .WithRequiredParameter("subscriptionId", LanguageConstants.String, "The subscription ID of the deployed availability zones")
+                .WithRequiredParameter("location", LanguageConstants.String, "The location of the availability zone mappings")
+                .WithRequiredParameter("physicalZone", LanguageConstants.String, "The physical zone to convert")
+                .Build();
+            
+            yield return new FunctionOverloadBuilder("toLogicalZones")
+                .WithReturnType(LanguageConstants.Array)
```

Comment:
If you can be more specific about the return type here, that would be helpful if the result of an invocation is itself passed to another function (like `first(toLogicalZones(<args>))`).

Take a look at the return type of `split` (line 268) to see how to define a more specific array return type. For example, if this function returns an array of strings and will always have at least one member, you can define the return type as `new TypedArrayType(LanguageConstants.String, TypeSymbolValidationFlags.Default, minLength: 1)`. With that definition, the return type of `first(toLogicalZones(<args>))` is `string`; with the current definition, the return type of `first(toLogicalZones(<args>))` is `any`.

Reply:
Thank you for the detailed explanation. The specific example you listed was very helpful the exact case I needed, an array of strings.